packages/cli/src/commands/gsn-send-request.ts

```diff
@@ -0,0 +1,144 @@
+  .option('--methodParams <items>', 'comma separated args list', commaSeparatedList)
+  .option('--calldata <string>', 'exact calldata to use')
+  .option('--to <string>', 'target RelayRecipient contract')
+  .option('--paymaster <string>', 'the Paymaster contract to be used')
+  .parse(process.argv)
+
+async function getProvider (
+  to: Address,
+  paymaster: Address,
+  mnemonic: string | undefined,
+  logger: LoggerInterface,
+  host: string): Promise<{ provider: Web3ProviderBaseInterface, from: Address }> {
+  const config: Partial<GSNConfig> = {
+    clientId: '0',
+    paymasterAddress: paymaster
+  }
+  const provider = new Web3.providers.HttpProvider(host, {
+    keepAlive: true,
+    timeout: 120000
+  })
+  let from: Address
+  let privateKey: PrefixedHexString | undefined
+  if (commander.from != null) {
+    // provider-controlled private key
+    from = commander.from
+    console.log('using', from)
+  } else if (mnemonic != null) {
+    const hdwallet = EthereumHDKey.fromMasterSeed(
+      bip39.mnemonicToSeedSync(mnemonic)
+    )
+    // add mnemonic private key to the account manager as an 'ephemeral key'
+    const wallet = hdwallet.deriveChild(0).getWallet()
+    from = `0x${wallet.getAddress().toString('hex')}`
+    privateKey = `0x${wallet.getPrivateKey().toString('hex')}`
+    console.log('mnemonic account:', from)
+  } else {
+    throw new Error('must specify either "--mnemonic" or pass "--from" account')
+  }
+  if (commander.directCall === true) {
+    return { provider, from }
+  } else {
+    if (paymaster == null) {
+      throw new Error('--paymaster: address not specified')
+    }
+    const overrideDependencies: Partial<GSNDependencies> = {
+      logger
+    }
+    const input: GSNUnresolvedConstructorInput = {
+      provider,
+      config,
+      overrideDependencies
+    }
+    const relayProvider = await RelayProvider.newProvider(input).init()
+    if (privateKey != null) {
+      relayProvider.relayClient.dependencies.accountManager.addAccount(privateKey)
+    }
+    return {
+      provider: relayProvider,
+      from
+    }
+  }
+}
+
+(async () => {
+  const network: string = commander.network
+  const nodeURL = getNetworkUrl(network)
+  const logger = createCommandsLogger(commander.loglevel)
+  const mnemonic = getMnemonic(commander.mnemonic)
+  const logic = new CommandsLogic(nodeURL, logger, {}, mnemonic)
+  const { provider, from } = await getProvider(
+    commander.to,
+    commander.paymaster,
+    mnemonic,
+    logger,
+    nodeURL
+  )
+  if (commander.abiFile == null || !fs.existsSync(commander.abiFile)) {
+    const file: string = commander.abiFile
+    throw new Error(`--abiFile: ABI file ${file} does not exist`)
+  }
+  const abiJson = JSON.parse(fs.readFileSync(commander.abiFile, 'utf8'))
+  if (commander.to == null) {
+    throw new Error('--to: target address is missing')
+  }
+  const web3Contract = logic.contract(abiJson, commander.to)
```

Comment:
you require the abi file to be ABI only: can't pass "artifact" file, which contains `{ abi: [] }` inside it
I think we should support both: either an array of items ( abiJson[0].type ) or artifact file where `.abi` is the array.